serial/uart.c

```diff
@@ -81,12 +132,20 @@ uart_config(uint8 uart_no)
     // to set the threshold here...
     // We do not enable framing error interrupts 'cause they tend to cause an interrupt avalanche
     // and instead just poll for them when we get a std RX interrupt.
+
+	uint32_t tx_empty_bits = 0;
+	if (uart0_tx_enable_pin >= 0) {
+	  // Set the empty threshold to 0 and enable the buffer empty interrupt
+	  tx_empty_bits = (0 & UART_TXFIFO_EMPTY_THRHD) << UART_TXFIFO_EMPTY_THRHD_S | UART_TXFIFO_EMPTY_INT_ENA;
+	}
     WRITE_PERI_REG(UART_CONF1(uart_no),
                    ((80 & UART_RXFIFO_FULL_THRHD) << UART_RXFIFO_FULL_THRHD_S) |
                    ((100 & UART_RX_FLOW_THRHD) << UART_RX_FLOW_THRHD_S) |
                    UART_RX_FLOW_EN |
                    (4 & UART_RX_TOUT_THRHD) << UART_RX_TOUT_THRHD_S |
-                   UART_RX_TOUT_EN);
+                   UART_RX_TOUT_EN) |
+                   tx_empty_bits
+         ;
     SET_PERI_REG_MASK(UART_INT_ENA(uart_no), UART_RXFIFO_FULL_INT_ENA | UART_RXFIFO_TOUT_INT_ENA);
```

Comment:
If I change `SET_PERI_REG_MASK(UART_INT_ENA(uart_no), UART_RXFIFO_FULL_INT_ENA | UART_RXFIFO_TOUT_INT_ENA | UART_TXFIFO_EMPTY_INT_ENA);` then program continuously going to line 282 and make error "unable run/load timer" or some like this. After few second watch dog make reset.